persist/postgres_persister.go

```diff
@@ -279,3 +279,133 @@ func (p PostgresPersister) DeleteCollection(ctx context.Context, id string) erro
 	_, err = p.db.ExecContext(ctx, "DELETE FROM collection WHERE id = $1", dbID)
 	return translateError(err)
 }
+
+// Post persistence methods
+
+// SelectPosts selects all posts
+func (p PostgresPersister) SelectPosts(ctx context.Context) ([]model.Post, error) {
+	rows, err := p.db.QueryContext(ctx, "SELECT id, collection_id, body, insert_time, last_update_time FROM post")
+	if err != nil {
+		return nil, translateError(err)
+	}
+	defer rows.Close()
+	posts := make([]model.Post, 0)
+	for rows.Next() {
+		var dbID int32
+		var collectionID int32
+		var post model.Post
+		err := rows.Scan(&dbID, &collectionID, &post.PostBody, &post.InsertTime, &post.LastUpdateTime)
+		if err != nil {
+			return nil, translateError(err)
+		}
+		post.ID = model.MakePostID(dbID)
+		post.Collection = model.MakeCollectionID(collectionID)
+		posts = append(posts, post)
+	}
+	return posts, nil
+}
+
+// SelectOnePost selects a single post
+func (p PostgresPersister) SelectOnePost(ctx context.Context, id string) (model.Post, error) {
+	var post model.Post
+	var dbID int32
+	var catID int32
+	n, err := fmt.Sscanf(id+"\n", p.pathPrefix+model.PostIDFormat+"\n", &dbID)
+	if err != nil || n != 1 {
+		// Bad ID
+		return post, model.NewError(model.ErrNotFound, id)
+	}
+	err = p.db.QueryRowContext(ctx, "SELECT id, collection_id, body, insert_time, last_update_time FROM post WHERE id=$1", dbID).Scan(
+		&dbID,
+		&catID,
+		&post.PostBody,
+		&post.InsertTime,
+		&post.LastUpdateTime,
+	)
+	if err != nil {
+		return post, translateError(err)
+	}
+	post.ID = model.MakePostID(dbID)
+	post.Collection = model.MakeCollectionID(catID)
+	return post, nil
+}
+
+// InsertPost inserts a PostBody into the database and returns the inserted Post
+func (p PostgresPersister) InsertPost(ctx context.Context, in model.PostIn) (model.Post, error) {
+	var dbID int32
+	var post model.Post
+	var catID int32
+	n, err := fmt.Sscanf(in.Collection, p.pathPrefix+model.CollectionIDFormat, &catID)
```

Comment:
I think this line should be 
```
n, err := fmt.Sscanf(in.Collection+"\n", p.pathPrefix+model.CollectionIDFormat+"\n", &catID)
```
(I had it wrong in the category reference code you copied it from.)